docs/timezone-draft.md

```diff
@@ -0,0 +1,160 @@
+# Draft design of custom time zone API
+
+## Why?
+
+Currently, `Temporal.TimeZone` objects can only be obtained by passing an identifier to the constructor or to `Temporal.TimeZone.from()`.
+These identifiers can be:
+
+- an offset string such as `-07:00`, in which case a time zone object is returned with a constant offset, with no daylight saving changes ever;
+- an IANA time zone name such as `America/New_York`, in which case a time zone object is returned that will give a different offset depending on the date that it is queried with, which attempts to be historically accurate (within reason) as to the actual time that clocks showed there.
+
+No other identifiers than the ones deemed valid by one of the above two rules (`±HH:MM` or [IsValidTimeZoneName](https://tc39.es/ecma402/#sec-isvalidtimezonename)), are currently permitted.
+
+Custom time zones are a desirable feature because implementations may want to prevent leaking information about what historical revision of the time zone data is present.
+API users may also want to perform time zone calculations with older (or newer!) versions of the time zone data than what is present in their implementation.
+
+## Usage of a custom time zone
+
+Like built-in time zones, custom time zones have an `[[Identifier]]`.
+In these examples we assume a custom time zone class `MarsTime` with the identifier `Mars/Olympus_Mons`.
```

Comment:
Mars time was meant as a toy example, but I'm not sure I can come up with a more compelling use case that's not locking down information exposed from the host system. Maybe someone else has a good idea?

Reply:
A use could be if you want to express the time in a location predating the railroad; for example, maybe you want "St. Louis Time in 1800", which would have been calibrated to the sun rather than to GMT.

Reply:
I managed to discover in a 120-year-old astronomy manual that St. Louis local mean time is GMT-06:00:49.11... isn't the internet great?! :smile: